packages/logging_cloudwatch/amplify_logging_cloudwatch/lib/src/queued_item_store/dart_queued_item_store.web.dart

```diff
@@ -15,18 +17,23 @@ class DartQueuedItemStore
     implements QueuedItemStore, Closeable {
   /// {@macro amplify_logging_cloudwatch.index_db_queued_item_store}
   // ignore: avoid_unused_constructor_parameters
-  DartQueuedItemStore(String? storagePath);
+  DartQueuedItemStore(String? storagePath) {
```

Comment:
Shouldn't this be an async constructor?  Also doesn't this introduce race conditions if you call the constructor and then call one of the async instance methods?  The instance methods do not check if _init() was completed.

Reply:
it should not cause race conditions because Dart event loop handles microtasks queue first and then events queue. 
to do it async and wait for _database to init means all instance methods needs to be async. currently the `isFull` method is a sync method. can make all instance method async and wait for _database if that's the right way to do it.

Reply:
Thanks that makes a lot more sense then.  We haven't really used microtasks in the code before and I'm unfamiliar with them but that makes sense.  

But yes, the async Future methods will be good but sync method `isFull` will likely not work.

Reply:
make sense. I will update the code to make the _database be a Future<QueuedItemStore> and the isFull to be async.